diff --git a/dist/auth/strategies/local/authenticate.js b/dist/auth/strategies/local/authenticate.js
index 86ac1733797fdbd53debdcb1c12ffd20ee847d1f..cb59caf4f85c2dcaa6c12ad3f2a826d0b8b1a94f 100644
--- a/dist/auth/strategies/local/authenticate.js
+++ b/dist/auth/strategies/local/authenticate.js
@@ -1,28 +1,25 @@
 // @ts-strict-ignore
 import crypto from 'crypto';
-import { getPasswordHashParameters, isCurrentPasswordHash } from './generatePasswordSaltHash.js';
+import { getPasswordHashParameters, isCurrentPasswordHash, pbkdf2Promisified } from './generatePasswordSaltHash.js';
 export const authenticateLocalStrategy = async ({ doc, password })=>{
     try {
         const { hash, salt } = doc;
         if (typeof salt === 'string' && typeof hash === 'string') {
             const { hash: storedHash, iterations, keyLength } = getPasswordHashParameters(hash);
-            const res = await new Promise((resolve, reject)=>{
-                crypto.pbkdf2(password, salt, iterations, keyLength, 'sha256', (e, hashBuffer)=>{
-                    if (e) {
-                        reject(e);
-                    }
-                    const storedHashBuffer = Buffer.from(storedHash, 'hex');
-                    if (hashBuffer.length === storedHashBuffer.length && crypto.timingSafeEqual(hashBuffer, storedHashBuffer)) {
-                        resolve({
-                            doc,
-                            shouldUpdatePasswordHash: !isCurrentPasswordHash(hash)
-                        });
-                    } else {
-                        reject(new Error('Invalid password'));
-                    }
-                });
+            const hashBuffer = await pbkdf2Promisified({
+                iterations,
+                keyLength,
+                password,
+                salt
             });
-            return res;
+            const storedHashBuffer = Buffer.from(storedHash, 'hex');
+            if (hashBuffer.length === storedHashBuffer.length && crypto.timingSafeEqual(hashBuffer, storedHashBuffer)) {
+                return {
+                    doc,
+                    shouldUpdatePasswordHash: !isCurrentPasswordHash(hash)
+                };
+            }
+            throw new Error('Invalid password');
         }
         return null;
     } catch (ignore) {
diff --git a/dist/auth/strategies/local/generatePasswordSaltHash.js b/dist/auth/strategies/local/generatePasswordSaltHash.js
index 7eb8b1c2e274e5c58736c7b29a77424ed91de0fc..8d0f02ad1fe3b4dfbc7300d0d8c1e0a40a94c9ef 100644
--- a/dist/auth/strategies/local/generatePasswordSaltHash.js
+++ b/dist/auth/strategies/local/generatePasswordSaltHash.js
@@ -68,8 +68,17 @@ export const isCurrentPasswordHash = (hash)=>typeof hash === 'string' && hash.st
 function randomBytes() {
     return new Promise((resolve, reject)=>crypto.randomBytes(32, (err, saltBuffer)=>err ? reject(err) : resolve(saltBuffer)));
 }
-function pbkdf2Promisified({ iterations, keyLength, password, salt }) {
-    return new Promise((resolve, reject)=>crypto.pbkdf2(password, salt, iterations, keyLength, 'sha256', (err, hashRaw)=>err ? reject(err) : resolve(hashRaw)));
+export async function pbkdf2Promisified({ iterations, keyLength, password, salt }) {
+    const passwordKey = await crypto.subtle.importKey('raw', Buffer.from(password, 'utf8'), 'PBKDF2', false, [
+        'deriveBits'
+    ]);
+    const hashRaw = await crypto.subtle.deriveBits({
+        name: 'PBKDF2',
+        hash: 'SHA-256',
+        salt: Buffer.from(salt, 'utf8'),
+        iterations
+    }, passwordKey, keyLength * 8);
+    return Buffer.from(hashRaw);
 }
 
 //# sourceMappingURL=generatePasswordSaltHash.js.map
